### PR TITLE
Create the persistence layer

### PR DESCRIPTION
**Pull request by malukenho**

### CODE REVIEW

File: tests/PhrasesTest/Persistance/ArrayTest.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ */
+class MemoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testFindOneRandomWithAnEmptyListReturnAnEmptyArray()
+    {
+        $list = array();
+        $phrases = new Memory($list);
+
+        $this->assertInstanceOf(
+            'Phrases\Persistance\RepositoryInterface',
```
Comment (malukenho) [inline tests/PhrasesTest/Persistance/ArrayTest.php]: @augustohp we can use `::class` operator here?

Comment (augustohp) [inline tests/PhrasesTest/Persistance/ArrayTest.php]: Sure 🍻

**Comment by malukenho** [comment]
@jackmakiyama @augustohp what is missing in this task?

**Comment by augustohp** [comment]
The controller saving phrases.

**Comment by malukenho** [comment]
@jackmakiyama @augustohp travis working :beers:

**Comment by malukenho** [comment]
@augustohp @jackmakiyama I need help to make tests run on travis. =x
captura de tela 2015-01-19 22 58 51

**Comment by jackmakiyama** [comment]
@malukenho we have a conflicts of features, please see [that](https://github.com/jackmakiyama/phrases/tree/feature/persist-the-phrases).
In my last commit I don't finish the implementation of the entity on relational layer and I finish tonight. How do we can resolve that?

**Comment by malukenho** [comment]
@jackmakiyama I think you can PR against my `feature/persist-the-phrases` branch, and we can solve conflicts.

/cc @augustohp

**Comment by jackmakiyama** [comment]
Okay, PR #21 made.

**Comment by malukenho** [comment]
@jackmakiyama closed #22 because it's merged here.

**Comment by malukenho** [comment]
@augustohp review?
In my view, can be merged :)

File: tests/PhrasesTest/ApplicationTest.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+     * @var Application
+     */
+    private $application;
+    private $persistance;
+
+    public function setUp()
+    {
+        $phrases = ConsumedData::asRelationalArray()[0];
+        $onePhrase = new Phrase($phrases['title'], $phrases['text']);
+        $this->persistance = new Memory([$onePhrase]);
```
Comment (jackmakiyama) [inline tests/PhrasesTest/ApplicationTest.php]: We need change this to Relational persistence mode.

**Comment by malukenho** [comment]
@jackmakiyama changed!

File: tests/PhrasesTest/ApplicationTest.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        unset($pdo);
+
+        $this->pdo = new Pdo('mysql:host=localhost;dbname=phrases_test', $username, $password);
+        $sql = 'CREATE TABLE IF NOT EXISTS phrases (
+            id INTEGER(11) PRIMARY KEY AUTO_INCREMENT,
+            title VARCHAR(255) NOT NULL UNIQUE,
+            text TEXT NOT NULL
+        ) Engine=InnoDB';
+
+        $this->pdo->exec($sql);
```
Comment (jackmakiyama) [inline tests/PhrasesTest/ApplicationTest.php]: @malukenho, what do you think to use the method [createTables](https://github.com/malukenho/phrases/pull/19/files#diff-b294ef2ee6c3f4ba61ef454ebe5ce813R7)?

File: features/createPhrase.feature [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Feature: akasdaks
```
Comment (augustohp) [inline features/createPhrase.feature]: :bomb:

File: tests/PhrasesTest/ApplicationTest.php (hunk 3 above)
Comment (augustohp) [inline tests/PhrasesTest/ApplicationTest.php]: :+1:

File: index.php [hunk 5]
```diff
@@ -1,9 +1,12 @@
...
-
 use Phrases\Application;
+use Phrases\Persistence;
+use Phrases\Http\Response\Send;
 
-chdir(__DIR__);
+require __DIR__ . '/vendor/autoload.php';
 
-require 'vendor/autoload.php';
+$persistance = new Persistence\Memory(['Jack Makiyama']);
```
Comment (augustohp) [inline index.php]: Acho que usar o SQLite é mais bacana, não? http://docs.travis-ci.com/user/languages/php/
Talvez salver o arquivo do banco no mesmo diretório da aplicação como padrão...

**Comment by augustohp** [comment]
:boom: No merge ae hein. :scream:

File: index.php (hunk 5 above)
Comment (jackmakiyama) [inline index.php]: :+1:

Comment (jackmakiyama) [inline index.php]: Os testes não deveriam quebrar com esse Memory aí?
Por isso que eu não mexi ontem.

File: tests/PhrasesTest/ApplicationTest.php (hunk 3 above)
Comment (jackmakiyama) [inline tests/PhrasesTest/ApplicationTest.php]: Done!

**Comment by malukenho** [comment]
@augustohp eu tinha feito o rebase, mas o @jackmakiyama desfez, e deixou em outra branch.

**Comment by jackmakiyama** [comment]
Desfiz pq estava quebrando os testes, acho melhor resolver no merge para a master mesmo do que fazer reebase.

**Comment by malukenho** [comment]
@jackmakiyama  :+1: , só pro augusto ficar sabendo :)

**Comment by jackmakiyama** [comment]
:smile:

File: index.php (hunk 5 above)
Comment (malukenho) [inline index.php]: @jackmakiyama a `index.php` não é um arquivo testado.

Comment (jackmakiyama) [inline index.php]: @malukenho mas e esse teste aqui: https://github.com/malukenho/phrases/pull/19/files#diff-dee727992e9c35e7a6540bbd21ca4470R62 ?
Acho que não saquei algo...

**Comment by jackmakiyama** [comment]
:beers:

**Pull request merged**